OpenRA.Mods.Cnc/Traits/Disguise.cs

```diff
@@ -211,14 +223,58 @@ void HandleDisguise(Player oldEffectiveOwner, bool oldDisguiseSetting)
 
 		void INotifyAttack.Attacking(Actor self, Target target, Armament a, Barrel barrel)
 		{
-			if (info.RevealDisguiseOn.HasFlag(RevealDisguiseType.Attack))
+			if (info.RevealDisguiseOn.HasFlag(RevealDisguiseType.Attack) || info.RevealDisguiseOn.HasFlag(RevealDisguiseType.Demolish))
```

Comment:
hmm didn't notice that particular interface to use, could swear I looked through the list of interfaces trying to find it.  In any event that solves that mystery so it'll get fixed. :)